### DISCUSSION TITLE
Which function or code is for calculating the highlighting parts of the left and right lines?

### DISCUSSION THREAD

**Discussion by roadfar** [dandavison/delta#1914] [General]
As lines 5, 6, 13, and 23 show in the following figure show, which function or code is for calculating the highlighting parts (i.e., the string start index and end index) of the left and right lines?
![image](https://github.com/user-attachments/assets/e2066e16-f46c-472c-8275-0750a9b98267)

**Comment by dandavison** [comment]
Hi @roadfar, I'm not sure I understand the question exactly. You're asking about delta implementation, right? Have you looked at https://github.com/dandavison/delta/blob/main/ARCHITECTURE.md?

**Reply by roadfar** [reply]
Hi, @dandavison  Thanks for your reply.
Yes. I am also writing a function that your project seems has implemented. Could you tell me which function or method in delta corresponds to the highlighting diff parts feature?
Thanks a lot.

**Reply by dandavison** [reply]
Have you read the ARCHITECTURE.md file I linked above?
